src/mcts/node.cc

```diff
@@ -402,8 +472,14 @@ bool NodeTree::ResetToPosition(const std::string& starting_fen,
   // previously trimmed; we need to reset current_head_ in that case.
   // Also, if the current_head_ is terminal, reset that as well to allow forced
   // analysis of WDL hits, or possibly 3 fold or 50 move "draws", etc.
-  if (!seen_old_head || current_head_->IsTerminal()) TrimTreeAtHead();
-
+  if (!seen_old_head) TrimTreeAtHead(); // must check if edges are created
+
+  // Certainty Propagation: No need to trim the head, just resetting certainty 
+  // state and recomputing N should suffice even with WDL hits. 
+  // TODO: Works, but maybe recheck if there are crititcal edge cases.
+  if (current_head_->GetParent())
```

Comment:
Why is it needed here? MakeMove already does it.

Reply:
Not quite sure what you mean?